scripts/deploy-federation-with-helm-chart.sh

```diff
@@ -0,0 +1,130 @@
+# hosting cluster as a member of the federation.
+#
+# WARNING: The service account for the federation namespace will be
+# granted the cluster-admin role.  Until more restrictive permissions
+# are used, access to the federation namespace should be restricted to
+# trusted users.
+#
+# If using minikube, a cluster must be started prior to invoking this
+# script:
+#
+#   $ minikube start
+#
+# This script depends on helm and kubectl being installed in
+# the path.  These and other test binaries can be installed via the
+# download-binaries.sh script, which downloads to ./bin:
+#
+#   $ ./scripts/download-binaries.sh
+#   $ export PATH=$(pwd)/bin:${PATH}
+#
+# To redeploy federation from scratch, prefix the deploy invocation with the deletion script:
+#
+#   # WARNING: The deletion script will remove federation and cluster registry data
+#   $ ./scripts/delete-federation-with-helm-chart.sh [join-cluster]... && ./scripts/deploy-federation-with-helm-chart.sh <image> [join-cluster]...
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+source "$(dirname "${BASH_SOURCE}")/util.sh"
+set -x
+
+NS="${FEDERATION_NAMESPACE:-federation-system}"
+PUBLIC_NS=kube-multicluster-public
+IMAGE_NAME="${1:-}"
+SYNC_CHART="${SYNC_CHART:-}"
+
+LATEST_IMAGE_NAME=quay.io/kubernetes-multicluster/federation-v2:latest
+USE_LATEST=
+if [[ "${IMAGE_NAME}" == "$LATEST_IMAGE_NAME" ]]; then
+  USE_LATEST=y
+fi
+
+KF_NS_ARGS="--federation-namespace=${NS} "
+
+if [[ -z "${IMAGE_NAME}" ]]; then
+  >&2 echo "Usage: $0 <image> [join-cluster]...
+
+<image>        should be in the form <containerregistry>/<username>/<imagename>:<tagname>
+
+Example: docker.io/<username>/federation-v2:test
+
+If intending to use the docker hub as the container registry to push
+the federation image to, make sure to login to the local docker daemon
+to ensure credentials are available for push:
+
+  $ docker login --username <username>
+
+<join-cluster> should be the kubeconfig context name for the additional cluster to join.
+               NOTE: The host cluster is already included in the join.
+
+"
+  exit 2
+fi
+
+shift
+JOIN_CLUSTERS="${*}"
+
+# Use DOCKER_PUSH=false ./scripts/deploy-federation.sh <image> to skip docker
+# push on container image when not using latest image.
+DOCKER_PUSH=${DOCKER_PUSH:-true}
+DOCKER_PUSH_CMD="docker push ${IMAGE_NAME}"
+if [[ "${DOCKER_PUSH}" == false ]]; then
+    DOCKER_PUSH_CMD=
+fi
+
+if [[ ! "${USE_LATEST}" ]]; then
+  base_dir="$(cd "$(dirname "$0")/.." ; pwd)"
+  dockerfile_dir="${base_dir}/images/federation-v2"
+  CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o "${dockerfile_dir}"/hyperfed "${base_dir}"/cmd/hyperfed/main.go
+  docker build ${dockerfile_dir} -t "${IMAGE_NAME}"
+  ${DOCKER_PUSH_CMD}
+fi
+
+if ! kubectl get ns "${PUBLIC_NS}" > /dev/null 2>&1; then
+    kubectl create ns "${PUBLIC_NS}"
+fi
+
+go build -o bin/kubefed2 cmd/kubefed2/kubefed2.go
+if [[ "${SYNC_CHART}" ]]; then
```

Comment:
When SYNC_CHART will be true?

Reply:
It is a function to update files before helm install.